src/graph_notebook/magics/graph_magic.py

```diff
@@ -107,6 +111,19 @@ def query_type_to_action(query_type):
         return 'sparqlupdate'
 
 
+def set_request_metrics(metadata: Metadata, time_elapsed: timedelta = None, status: int = None, status_ok: str = None,
```

Comment:
Can we move this into the `Metadata` object itself and take in a `requests.Response` object?

Reply:
Done